### ISSUE TITLE
GunBot v8.0.3 Incorrectly Averaging Buy Prices (Screenshot), Please Help.

### ISSUE THREAD

**Issue by pookzta** [GuntharDeNiro/BTCT#521]
Hi Gunthy Developers and Users,

GunBot v8.0.3 is definitely incorrectly averaging buy prices sometimes. It has resulted in some minor sales for losses, even with a protective gain setting, and it often misses opportunities to sell for profit because it thinks the sell price is not high enough, due to miscalculation of the average buy price. For some reason it appears to be averaging too many, or all, of the previous buy prices, rather than just the buy prices of the current bag. GunBot refused to sell an XLM bag recently because: **"Averaged bought price is: 0.000043693714349376117, waiting for sell price to grow"**, but if you look at the screenshot below, you will see that the _true_ average buy price is much lower than that, around _.00003100._ This is happening with the included GunBot strategies (e.g. BB, GAIN, etc.) and with the TradingView plugin. Please see the screenshot below and calculate the average buy price for yourself.

Thank you very much for any time, help, and comments that people contribute to this problem. Best wishes always.

-Abe 

![gunthyfalseavgbuyprice](https://user-images.githubusercontent.com/37188378/37169802-d1512c14-22bd-11e8-956b-0a904dd7f7ca.jpg)

**Comment by pookzta** [comment]
Temporary solution until this issue gets fixed:

Since v7.0.2 has a problem buying due to incorrect calculations of bag quantities (error was fixed in v8), I use v7 just to sell. 
Since v8.0.3 has a problem selling due to incorrect averaging of buy prices (error not present in v7), I use v8 just to buy.

Thus, a temporary solution, is to run a v7 instance on one computer (or virtual machine) with impossible buy parameters (max buy 0.0 BTC, max cap 0.0 BTC), so that it will only be able to sell, and to run a v8 instance on a second computer (or virtual machine) with impossible sell parameters (gain 99%, max sell 0.0 BTC), so that it will only really attempt to buy. 

In other words...

Run v7 just to sell, and v8 just to buy. Working great this way.

**Comment by markcaunter** [comment]
I've experienced this too and can add some details.  What I've derived is this:

When I have a coin on Binance that has dust, the dust causes the bot to calculate the average buy price based on the current buy price and the previous buy price.  This happens irrespective of what the MIN_VOLUME_TO_BUY value is set to.

This is a real life example:

1. The bot purchases 15 ELF at 0.00007567
2. The bot sells, but 0.985 ELF dust is left.

3. The bot purchases 16 ELF at 0.00007319
4. The bot reports a holding of 16.985 ELF at an average of 0.00007439

What it has incorrectly done is calculate the average like this:

((15 * 0.00007567) + (0.00007319 * 16)) / 31
= 0.00007439

**Comment by jasonyslai** [comment]
Omg, the bot counts the history avg price and get wrong avg price.
it can't make any sell now due to a ridiculous avg price.....

**Comment by jasonyslai** [comment]
Thanks for the support team.
This issue can be solved by overriding the parameter temporarily.
If your balance still have some dust, calculate the avg price by yourself and override the bought price:
"BTC-ONT": 	{"strategy": "bbstepgain","override": {"BOUGHT_PRICE":0.000496}}
To prevent another history trading effects the avg price, just reset it to zero:
"BTC-EOS": 	{"strategy": "tssl","override": {"BOUGHT_PRICE":0}}

You might need to consider a new bought price for each of pairs if you restart the apps.

**Comment by pookzta** [comment]
Gunthar has officially acknowledged that Gunbot v9 will fix this problem, and that the release is now in beta testing phase. Not long now folks, can't wait for v9!

**Comment by lhilarides** [comment]
So the workaround until v9 is to only trade "whole coin" pairs?
